Gems/HttpRequestor/Code/Source/HttpRequestManager.cpp

```diff
@@ -150,7 +155,9 @@ namespace HttpRequestor
             httpRequest->SetContentLength(AZStd::to_string(httpRequestParameters.GetBodyStream()->str().length()).c_str());
         }
 
-        auto httpResponse = httpClient->MakeRequest(httpRequest);
+        AZStd::chrono::steady_clock::time_point start = AZStd::chrono::steady_clock::now();
+        const auto httpResponse = httpClient->MakeRequest(httpRequest);
+        m_lastRoundTripTime = AZStd::chrono::duration_cast<AZStd::chrono::milliseconds>(AZStd::chrono::steady_clock::now() - start);
```

Comment:
Just double-checking, since i see a bit of threading code in this file, only one request / last round trip time happens at a time, correct? Just making sure there aren't any race conditions or anything on m_lastRoundTripTime.

Reply:
Also, assuming this is correct, might want to consider adding a comment above the places that set m_lastRoundTripTime explicitly calling out this assumption, so that if it ever changes, the ramifications will be more obvious.

Reply:
Correct, there's a separate single thread which makes the actual HTTP requests one at a time. During each request, m_lastRoundTripTime is set and the callback is called from that thread, which will also be when lastRoundTripTime is read.
 
There's a lock in that separate thread when it's altering the request queue since that thread will be be popping off the queue potentially while component in main thread are making new request (adding onto the queue)